### PR TITLE
Feat: added support for rego files to filter input messages

### PR DESCRIPTION
**Pull request by DmitriyLewen**
Issue 175

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,9 +111,18 @@ Key | Description | Possible Values | Example
 --- | --- | --- | ---
 *name*|Unique name of route| string | teams-vul-route
 *input*|A Rego rule to match against incoming messages. If there is a match then this route will be chosen for the incoming message| Rego language statements | contains(input.message,"alpine")
+*input-files*|One or more files with Rego rules| Set of Rego language files | ["route1.rego", "route2.rego"]
```
Comment (AndreyLevchenko) [inline README.md:114]: 'route' naming is no very good for example

File: rego-filters/Ignore-Registry.rego [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package postee
+
+
+ArrayIgnoreRegistry := {"all", "Aqua"} #Comma separated list of registries that will be ignored by the integration
```
Comment (AndreyLevchenko) [inline rego-filters/Ignore-Registry.rego:4]: some real life names are required

File: rego-filters/Igrore-Image-Name.rego [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package postee
+
+
+ArrayIgnoredImageNames := {"ignor"} #List of comma separated images that will be ignored by the integration
```
Comment (AndreyLevchenko) [inline rego-filters/Igrore-Image-Name.rego:4]: pick some image names from dockerhub, something like `alpine`

File: regoservice/regocheck.go [hunk 4]
```diff
@@ -3,31 +3,55 @@ package regoservice
...
+	if len(files) != 0 {
+		if os.Getenv("REGO_FILTERS_PATH") != "" {
+			pathToRegoFilters = os.Getenv("REGO_FILTERS_PATH")
+		}
+		if !strings.HasSuffix(pathToRegoFilters, "/") {
+			pathToRegoFilters += "/"
+		}
+		for i, file := range files {
+			if !strings.HasPrefix(file, pathToRegoFilters) {
+				files[i] = pathToRegoFilters + file
```
Comment (AndreyLevchenko) [inline regoservice/regocheck.go:38]: use `filepath.Join()` to join parts of the path

File: ui/frontend/src/components/RouteDetails.vue [hunk 5]
```diff
@@ -235,6 +246,15 @@ export default {
       }
       this.formValues.plugins[propName] = v;
     },
+    updateInputFilesCollectionField(e) {
```
Comment (AndreyLevchenko) [inline ui/frontend/src/components/RouteDetails.vue:249]: it's copy of `updateCollectionField()`. Also why `!this.formValues` is required

File: regoservice/regocheck.go [hunk 6]
```diff
@@ -3,31 +3,55 @@ package regoservice
...
+func DoesMatchRegoCriteria(input interface{}, files []string, rule string) (bool, error) {
+	ctx := context.Background()
+	r := &rego.Rego{}
+	pathToRegoFilters := defaultPathToRegoFilters
 
-	if rule == "" {
-		return true, nil //no rule defined - any input allowed
-	}
+	if len(files) != 0 {
+		if os.Getenv("REGO_FILTERS_PATH") != "" {
```
Comment (AndreyLevchenko) [inline regoservice/regocheck.go:30]: call `os.Getenv()` once

File: regoservice/regocheck.go (hunk 4 above)
Comment (AndreyLevchenko) [reply regoservice/regocheck.go:38]: also move all path merging logic to separate function

File: regoservice/regocheck.go [hunk 7]
```diff
@@ -3,31 +3,55 @@ package regoservice
...
+		}
+		if !strings.HasSuffix(pathToRegoFilters, "/") {
+			pathToRegoFilters += "/"
+		}
+		for i, file := range files {
+			if !strings.HasPrefix(file, pathToRegoFilters) {
+				files[i] = pathToRegoFilters + file
+			}
+		}
+		r = rego.New(
```
Comment (AndreyLevchenko) [inline regoservice/regocheck.go:41]: call `rego.New(...)` once. Use something like `func buildRegoLoader() func(r *rego.Rego) ..` to provide different loaders

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (DmitriyLewen) [reply README.md]: changed

File: rego-filters/Ignore-Registry.rego (hunk 2 above)
Comment (DmitriyLewen) [reply rego-filters/Ignore-Registry.rego]: changed

File: rego-filters/Igrore-Image-Name.rego (hunk 3 above)
Comment (DmitriyLewen) [reply rego-filters/Igrore-Image-Name.rego]: done

File: regoservice/regocheck.go (hunk 4 above)
Comment (DmitriyLewen) [reply regoservice/regocheck.go]: done

File: ui/frontend/src/components/RouteDetails.vue (hunk 5 above)
Comment (DmitriyLewen) [reply ui/frontend/src/components/RouteDetails.vue:262]: removed this check

File: regoservice/regocheck.go (hunk 6 above)
Comment (DmitriyLewen) [reply regoservice/regocheck.go:29]: done

File: regoservice/regocheck.go (hunk 7 above)
Comment (DmitriyLewen) [reply regoservice/regocheck.go]: done

**Pull request merged**